packages/client/src/stores/ReportStore.ts

```diff
@@ -52,26 +52,28 @@ export default class ReportStore {
     this.thresholds = (canaryExecutionStatusResponse.canaryExecutionRequest as CanaryExecutionRequest).thresholds;
     this.metricSetPairListId = canaryExecutionStatusResponse.metricSetPairListId as string;
 
-    const judgeResult = ofNullable(this.result.judgeResult).orElseThrow(
-      () => new Error('Expected there to be judgement results on the Canary Result object')
-    );
+    safeGet(() => this.result.judgeResult).ifPresent(judgeResultFromResult => {
+      const judgeResult = judgeResultFromResult;
 
-    judgeResult.results.forEach((canaryAnalysisResult: CanaryAnalysisResult) => {
-      this.canaryAnalysisResultByIdMap[canaryAnalysisResult.id] = canaryAnalysisResult;
+      judgeResult.results.forEach((canaryAnalysisResult: CanaryAnalysisResult) => {
+        this.canaryAnalysisResultByIdMap[canaryAnalysisResult.id] = canaryAnalysisResult;
+      });
     });
   }
 
   @computed
   get idListByMetricGroupNameMap(): KvMap<string[]> {
     const idListByMetricGroupNameMap: KvMap<string[]> = {};
 
-    this.result.judgeResult!.results.forEach(canaryAnalysisResult => {
-      canaryAnalysisResult.groups.forEach(group => {
-        if (idListByMetricGroupNameMap[group]) {
-          idListByMetricGroupNameMap[group].push(canaryAnalysisResult.id);
-        } else {
-          idListByMetricGroupNameMap[group] = [canaryAnalysisResult.id];
-        }
+    safeGet(() => this.result.judgeResult).ifPresent(judgeResult => {
```

Comment:
```
safeGet(this.result.judgeResult!.results).ifPresent(results => {
```

or

```
ofNullable(() => this.result.judgeResult).ifPresent(judgeResult => {
```

Reply:
I might be confused, but the thing that wasn't guaranteed before with the `!` was the `results`

`safeGet` lets you chain into some object structure where 1 or more things could yield a NPE and give you an Optional of the thing and swallow any NPEs.

`ofNullable` doesn't swallow NPEs and is for just single level depth things.